custom_components/ha-google-home/api.py

```diff
@@ -52,18 +52,29 @@ def get_android_id(self):
         """Generate random android_id"""
         return self._client.get_android_id()
 
+    @staticmethod
+    def format_offline_devices_to_human_string(device_list):
+        device_list = str(device_list)
+        char = "[']"
+        for c in char:
+            device_list = device_list.replace(c, "")
+
+        return device_list
+
     @staticmethod
     def create_url(ip, port, api_endpoint):
         """Creates url to endpoint.
         Note: port argument is unused because all request must be done to 8443"""
         url = "https://{ip}:{port}/{endpoint}".format(
-            ip=ip, port=str(PORT), endpoint=api_endpoint
+            ip=ip, port=str(port), endpoint=api_endpoint
         )
         return url
 
     def get_alarms_and_timers_from(self, device, endpoint):
         """Fetches timers and alarms from google device"""
-        url = self.create_url(device.ip, device.port, endpoint)
+        if device.ip is None:
+            return
```

Comment:
Can we do that check outside this method? It would make a bit more sense 
See comment below